fix(forms): Make some fixes to typed forms API and docs.

dylhunn opened the pull request:
* `FormRecord` JSDoc should now appear on a.io
* The `{@see foo#bar}` syntax previously did not work, and has been replace with backticks
* Value and RawValue have been removed from the public API (rolling back #45978); this needs more discussion (see #fw-forms)

dylhunn committed:
docs(forms): Make some fixes to typed forms docs:

* `FormRecord` jsdocs should now appear on a.io
* The `{@see foo#bar}` syntax previously did not work, and has been replace with backticks

AndrewKushnir commented on packages/forms/src/model/form_group.ts:
```diff
@@ -629,49 +625,53 @@ export class FormRecord<TControl extends AbstractControl<Value<TControl>, RawVal
    * Replace an existing control.
    *
-   * {@see FormGroup#setControl}
+   * See `FormGroup#setControl`
```
nit: I'd propose updating this line (and similar ones in this PR) a bit:
```suggestion
   * See `FormGroup#setControl` for additional information.
```

AndrewKushnir reviewed (commented):
@dylhunn thanks for the updates 👍 I've left one minor comment.

I also think that we should probably have 2 separate commits in this PR: one is a revert of the previous commit that made the `Value` and `RawValue` public and the second one is the docs update (and make it `docs(forms): ...`, so it's not included in the changeog). That should help the tooling to ensure that the initial "feat(forms)" commit is not present in a changelog (or at least that'd be visible to a caretaker would would assemble the changelog for v14.0.0 release).

dylhunn committed:
revert "fix(forms): Value and RawValue should be part of the public API."

As per discussion on #fw-forms, this reverts #45978 (although the more in-depth comments were kept).

AndrewKushnir reviewed (approved):
Thanks for addressing the feedback 👍 

Could you please also change the `fix(forms):` -> `docs(forms):` in the commit message?

dylhunn commented:
Done, thanks!

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

alxhub reviewed (approved):
Reviewed-for: public-api

dylhunn commented:
merge-assistance: failures are unrelated

jessicajaniuk commented:
This PR was merged into the repository by commit 038ba8adeae833135c9ba068cae3aa3859016272.
